pkg/render/node.go

```diff
@@ -891,49 +853,81 @@ func (c *nodeComponent) nodeEnvVars() []v1.EnvVar {
 			if v4pool.NodeSelector != "" {
 				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV4POOL_NODE_SELECTOR", Value: v4pool.NodeSelector})
 			}
-		} else {
-			// If there's also no IPv6 pool, set NO_DEFAULT_POOLS to "true".
-			if v6pool == nil {
-				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "NO_DEFAULT_POOLS", Value: "true"})
-			}
 		}
 
-		// Env based on IPv6 auto-detection and ippool configuration.
-		v6Method := getAutodetectionMethod(c.cr.Spec.CalicoNetwork.NodeAddressAutodetectionV6)
-		if v6Method != "" {
-			// IPv6 Auto-detection is enabled.
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "IP6", Value: "autodetect"})
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "IP6_AUTODETECTION_METHOD", Value: v6Method})
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_IPV6SUPPORT", Value: "true"})
-
-			// Set CALICO_ROUTER_ID to "hash" if IPv6 only.
-			if v4Method == "" {
-				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_ROUTER_ID", Value: "hash"})
-			}
-		} else {
-			// IPv6 Auto-detection is disabled.
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "IP6", Value: "none"})
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_IPV6SUPPORT", Value: "false"})
-		}
-
-		if v6pool != nil {
+		// Configure IPv6 pool.
+		if v6pool := GetIPv6Pool(c.cr.Spec.CalicoNetwork.IPPools); v6pool != nil {
 			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_CIDR", Value: v6pool.CIDR})
 
 			if v6pool.BlockSize != nil {
 				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_BLOCK_SIZE", Value: fmt.Sprintf("%d", *v6pool.BlockSize)})
 			}
 			if v6pool.NATOutgoing == operator.NATOutgoingDisabled {
-				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NAT_OUTGOING",
-					Value: "false"})
+				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NAT_OUTGOING", Value: "false"})
 			}
 			if v6pool.NodeSelector != "" {
-				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NODE_SELECTOR",
-					Value: v6pool.NodeSelector})
+				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NODE_SELECTOR", Value: v6pool.NodeSelector})
 			}
 		}
+
+	}
+
+	// Determine MTU to use. If specified explicitly, use that. Otherwise, set defaults based on an overall
+	// MTU of 1460.
+	ipipMtu := "1440"
+	vxlanMtu := "1410"
+	wireguardMtu := "1400"
+	if c.cr.Spec.CalicoNetwork.MTU != nil {
+		ipipMtu = strconv.Itoa(int(*c.cr.Spec.CalicoNetwork.MTU))
+		vxlanMtu = strconv.Itoa(int(*c.cr.Spec.CalicoNetwork.MTU))
+		wireguardMtu = strconv.Itoa(int(*c.cr.Spec.CalicoNetwork.MTU))
+	}
+	nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_VXLANMTU", Value: vxlanMtu})
+	nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_WIREGUARDMTU", Value: wireguardMtu})
+
+	// Configure whether or not BGP should be enabled.
+	if !bgpEnabled(c.cr) {
```

Comment:
I think we should have this wrapped in a check for `if c.cr.Spec.CNI.Type == operator.PluginCalico` and set CALICO_NETWORKING_BACKEND=none if not.

Reply:
> and set CALICO_NETWORKING_BACKEND=none if not.

This is intentionally VXLAN even for non-calico CNI plugins in this PR, since there is a known issue that node IPs are not populated on the node object if set to "none".

I guess this is really just a workaround for a bug in another part of the code, though.

Reply:
Does that mean there is a problem with other CNI plugins currently that have that value set to none? Or is this only an issue for the Encryption case where we want to use wireguard with a other CNI plugins?